apollo-router/src/plugins/demand_control/mod.rs

```diff
@@ -0,0 +1,62 @@
+//! Demand control plugin.
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt;
+
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::services::execution::BoxService;
+
+/// Algorithm for calculating the cost of an incoming query.
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema)]
+#[serde(deny_unknown_fields, rename_all = "lowercase")]
+pub(crate) enum CostCalculationAlgorithm {
+    /// A simple, statically-defined cost mapping for operations and types.
+    ///
+    /// Operation costs:
+    /// - Mutation: 10
+    /// - Query: 0
+    /// - Subscription 0
+    ///
+    /// Type costs:
+    /// - Object: 1
+    /// - Interface: 1
+    /// - Union: 1
+    /// - Scalar: 0
+    /// - Enum: 0
+    #[default]
+    Default,
```

Comment:
Lets rename this to `Basic`, and have it not defaulted. Users will have to specify the value explicitly for now.

Reply:
https://github.com/apollographql/router/pull/4820/commits/234bff20ba6141fb2dc2fe78101d86f6c7a01036